tests/extmod/tls_dtls.py

```diff
@@ -0,0 +1,53 @@
+# Test DTLS functionality including timeout handling
+
+try:
+    from tls import PROTOCOL_DTLS_CLIENT, PROTOCOL_DTLS_SERVER, SSLContext, CERT_NONE
+    import socket, io
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+
+class DummySocket(io.IOBase):
+    def __init__(self):
+        self.write_buffer = bytearray()
+        self.read_buffer = bytearray()
+
+    def write(self, data):
+        self.write_buffer.extend(data)
+        return len(data)
+
+    def readinto(self, buf):
+        l = min(len(self.read_buffer), len(buf))
+        if l == 0:
+            return None
+        buf[:l] = self.read_buffer[:l]
+        self.read_buffer = self.read_buffer[l:]
+        return l
+
+    def ioctl(self, req, arg):
+        return 0
```

Comment:
Is this ioctl method needed?  If so, for which `req`uests?  Probably it should be:
```py
if req == 4:  # MP_STREAM_CLOSE
    return 0
return -1
```
Which follows a3128f89ccb1d24d0ddaec7164d2fa8a101b8ac3

Reply:
I believe it's just close yes.